dashboard/app/controllers/followers_controller.rb

```diff
@@ -49,6 +67,41 @@ def remove
     )
   end
 
+  def remove_async
+    @section = Section.find_by_code params[:section_code]
+
+    if @section
+      f = Follower.where(section: @section.id, student_user_id: current_user.id).first
+    end
+
+    unless @section && f
+      render json: {
+        result: "error",
+        sections: current_user.sections_as_student.map(&:summarize)
+      }
+      return
+    end
+
+    @teacher = @section.user
+
+    authorize! :destroy, f
+    f.delete
+    # Though in theory required, we are missing an email address for many teachers.
+    if @teacher && @teacher.email.present?
+      FollowerMailer.student_disassociated_notify_teacher(@teacher, current_user).deliver_now
+    end
+    teacher_name = @teacher ? @teacher.name : I18n.t('user.deleted_user')
+    render json: {
+      result: "success",
+      notice: t(
+        'teacher.student_teacher_disassociated',
+        teacher_name: teacher_name,
+        section_code: params[:section_code]
+      ),
+      sections: current_user.sections_as_student.map(&:summarize)
+    }
+  end
```

Comment:
Unless there's a compelling reason to keep them here, please move these API actions into a new controller under `/api/v1`. See below